### ISSUE TITLE
Incorrect behaviour of "no-unused-vars" rule when variable defined outside of the loop

### ISSUE THREAD

**Issue by markelog** [eslint/eslint#6125]
**What version of ESLint are you using?**
2.9.0

**What parser (default, Babel-ESLint, etc.) are you using?**
default

**Please show your full configuration:**
```json
{
	"rules": {
		"no-unused-vars": "error"
	}
}
```
**What did you do? Please include the actual source code causing the issue.**
run eslint on the following code - 
```js
(function( obj ) {
	var name;

	for ( name in obj ) {
		return false;
	}
	return true;
});
```

**What did you expect to happen?**
No errors

**What actually happened? Please include the actual, raw output from ESLint.**
```sh
2:6  error  'name' is defined but never used  no-unused-vars
```

**Comment by mysticatea** [comment]
Thank you for this issue.

Closing as a duplicate of #2342.
Please track that issue.
